assemble.py

```diff
@@ -213,22 +244,28 @@ def get_args():
         p = Progress(len(targets))
         logger.info('Found {} Targets to assemble image for'.format(found_targets_number))
         for target in targets:
-            logger.info('Assembling image for {}, shortlist: {}'.format(target.name, args.app_shortlist))
+            logger.info('Assembling image for %s, shortlist: %s, encrypted: %s',
+                        target.name, args.app_shortlist, args.app_encrypted)
             subprog = Progress(3, p)
             image_file_path = factory_client.get_target_system_image(target, args.out_image_dir, subprog)
+            target.shortlist = args.app_shortlist
```

Comment:
Maybe if a new property of Target is introduced, e.g. `Target.enc_shortlist` then we could avoid many 'if' statements at the upper level, it would be just in the scope of Fetcher implementation? (Just guessing :)).

Reply:
that way is possible. It requires adding new logic to the fetcher to fetch both things into different locations. And then you have to add a different set of if's into the target-store so that store content and name it correctly. That was my initial approach and I was hitting a lot of friction with the fetcher and store modules. I don't mind trying it again if you think that's the way to go.

Reply:
>I don't mind trying it again if you think that's the way to go.

No, no need, I am pretty sure I thought it through and I have already forgotten the implementation details :).